tests/test_pyav.py

```diff
@@ -0,0 +1,329 @@
+        meta = plugin.metadata()
+        assert meta["profile"] == "High 4:4:4 Predictive"
+        assert meta["codec"] == "h264"
+
+        plugin.metadata(index=4)
+    print("")
+
+
+def test_properties(test_images: Path):
+    with iio.imopen(str(test_images / "cockatoo.mp4"), "r", plugin="pyav") as plugin:
+        with pytest.raises(IOError):
+            # subsampled format
+            plugin.properties(format="yuv420p")
+
+        with pytest.raises(IOError):
+            # components per channel differs
+            plugin.properties(format="nv24")
+
+        props = plugin.properties()
+        assert props.shape == (3, 720, 1280)
+        assert props.dtype == np.uint8
+        assert props.is_batch is False
+
+        props = plugin.properties(index=4, format="rgb48")
+        assert props.shape == (720, 1280, 3)
+        assert props.dtype == np.uint16
+        assert props.is_batch is False
+
+        props = plugin.properties(index=None)
+        assert props.shape == (280, 3, 720, 1280)
+        assert props.dtype == np.uint8
+        assert props.is_batch is True
+
+
+def test_video_format_to_dtype():
+    fake_formats = [
+        "vdpau",
+        "dxva2_vld",
+        "vaapi_moco",
+        "drm_prime",
+        "mediacodec",
+        "qsv",
+        "mmal",
+        "vulkan",
+        "vaapi_vld",
+        "d3d11",
+        "d3d11va_vld",
+        "videotoolbox_vld",
+        "vaapi_idct",
+        "opencl",
+        "cuda",
+        "xvmc",
+    ]
+
+    for name in video_format_names:
+        format = av.VideoFormat(name)
+
+        if name in fake_formats:
+            # should fail
+            with pytest.raises(ValueError):
+                _format_to_dtype(format)
+        else:
+            # should succseed
+            _format_to_dtype(format)
+
+
+def test_filter_sequence(test_images):
+    frames = iio.imread(
+        test_images / "cockatoo.mp4",
+        index=None,
+        plugin="pyav",
+        filter_sequence=[
+            ("drawgrid", "w=iw/3:h=ih/3:t=2:c=white@0.5"),
+            ("scale", {"size": "vga", "flags": "lanczos"}),
+            ("tpad", "start=3"),
+        ],
+    )
+
+    assert frames.shape == (283, 3, 480, 640)
+
+    frames = iio.imread(
+        test_images / "cockatoo.mp4",
+        index=None,
+        plugin="pyav",
+        filter_sequence=[
+            ("framestep", "5"),
+            ("scale", {"size": "vga", "flags": "lanczos"}),
+        ],
+    )
+
+    assert frames.shape == (56, 3, 480, 640)
+
+
+def test_write_bytes(test_images, tmp_path):
+    img = iio.imread(
+        test_images / "cockatoo.mp4",
+        index=None,
+        plugin="pyav",
+        filter_sequence=[
+            ("framestep", "5"),
+            ("scale", {"size": "vga", "flags": "lanczos"}),
+        ],
+    )
+    img_bytes = iio.imwrite(
+        "<bytes>",
+        img,
+        format_hint=".mp4",
+        plugin="pyav",
+        in_pixel_format="yuv444p",
```

Comment:
By default, pyav returns images in `yuv444p`?  Shouldn't you want things as `rgb` by default in a `python` world?

Reply:
@hmaarrfk By default, it returns the pixels in the same format as they were stored in the file. If we don't know how to return that format (e.g. because it's not trivally stridable), we raise an exception and ask the user to explicitly specify a suitable format via the `format=` kwarg, e.g., `format="rgb24"` to return RGB frames.

Reply:
understood. thank you.